webdav/pkg/dav/requests/thumbnail.go

```diff
@@ -0,0 +1,95 @@
+package requests
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/go-chi/chi"
+)
+
+const (
+	// DefaultWidth defines the default width of a thumbnail
+	DefaultWidth = 32
+	// DefaultHeight defines the default height of a thumbnail
+	DefaultHeight = 32
+)
+
+// ThumbnailRequest combines all parameters provided when requesting a thumbnail
+type ThumbnailRequest struct {
+	Filepath        string
+	Filename        string
+	Extension       string
+	Width           int32
+	Height          int32
+	PublicLinkToken string
+}
+
+// ParseThumbnailRequest extracts all required parameters from a http request.
+func ParseThumbnailRequest(r *http.Request) (ThumbnailRequest, error) {
+	fp := extractFilePath(r)
+	q := r.URL.Query()
+
+	width, height, err := parseDimensions(q)
+	if err != nil {
+		return ThumbnailRequest{}, err
```

Comment:
does it have any meaning to return an empty `ThumbnailRequest` or should it be equivalent to have a `*ThumbnailRequest` as a return argument and just stick to `return nil, err`?